libs/client.go

```diff
@@ -68,6 +75,73 @@ func NewClient() *Client {
 	return &client
 }
 
+// SetProxy adds a proxy to the client
+func (c *Client) SetProxy(proxyList []string) {
+list:
+	for _, p := range proxyList {
+		proxyurl, err := url.Parse(p)
```

Comment:
I think this code can be moved into a proper function since it's repeated at line 103